src/automation_agent/review_provider.py

```
@@ -40,114 +47,249 @@ class LLMReviewProvider(ReviewProvider):
+    async def review_code(self, diff: str) -> tuple[Union[str, Dict[str, Any]], Dict[str, Any]]:
         """Review code using LLM.
         
         Returns:
-            str: Review content on success
-            Dict[str, Any]: Error dict on rate limit or other failures
+            Tuple of (review_content_or_error_dict, usage_metadata)
         """
         try:
-            return await self.llm.analyze_code(diff)
+            review, metadata = await self.llm.analyze_code(diff)
+            return review, metadata
```

Ruff (0.14.7)

Consider moving this statement to an `else` block

(TRY300)